### PR TITLE
Allow openshift to work with docker and jib container images

### LINKED ISSUE #9663: Improved OpenShift Deployments
**Description**
I like the Kubernetes and OpenShift extensions since they generate the yaml files and if necessary you can change these settings easily, for example via properties file. I think that's the right approach. Make it simple for first time Quarkus developers, but allow them to change things later when needed. The same goes for the generated Dockerfiles, so that, for example, you can change to OpenJ9.

It's too bad though that you can only use S2I to deploy to OpenShift. In my case I used Java 11 locally which didn't work with the S2I standard image which only supports Java 8 since I had to compile locally rather than on the server. I could implement my own image builder, but why should I have to do this? I have my Dockerfile and yaml files exactly how I want them. Even if there were a Java 11 S2I, I'd prefer to use my own Dockerfile.

The alternatives have weaknesses for my requirements too: Binary builds allow you to build on the server side and to use existing Dockerfiles and yaml files, but I don't see how you can refer to existing (remote) Git repos. When you create builds from remote Git repos, you cannot apply your own yaml files. The Jib and Docker options I didn't get to work on my Mac even after I created a route to the registry.

I think it would be great to have solutions for two scenarios:

1) Developer wants to quickly push and test
The Quarkus extensions come very close to solving this already. Unfortunately S2I is the only option which means I cannot use my own Dockerfile and I'm forced to use an older Java version I haven't tested with.

2) Deployments to production environments
In this case the code is pushed to a Git repo first, tagged, etc. which means that mechanisms like binary builds cannot be used. Jenkins pipelines seem to be deprecated. I haven't seen documentation or best practices for how to do this.

**Implementation ideas**
For scenario (1): Could the Kubernetes/OpenShift extension be extended to use the Dockerfile in the project rather than using S2I?

For scenario (2) it seems like there is a discussion going on already: https://github.com/quarkusio/quarkus/issues/6147

### LINKED ISSUE #10055: OpenShift deployment with kubernetes & container-image-docker extension does not work
**Describe the bug**

The combination of both extensions **quarkus-kubernetes** and **quarkus-container-image-docker** for **deployment-target=openshift** does not work out of the box.

The generated openshift.yml file contains some not needed resources and wrong settings for S2I build (S2I is not in place in my configuration).

To get the deployment done I have to go the OpenShift WebConsole and manually delete the wrong deployment-trigger on the DeploymentConfig and start the deployment.

The following resources are generated by the quarkus-kubernetes extension:

1) ServiceAccount --> okay
2) Service --> okay
3) ImageStream as target for the S2I build --> **not needed (must be removed)**
4) ImageStream for S2I builder image --> **not needed (must be removed)**
5) BuildConfig for S2I build --> **not needed (must be removed)**
6) DeploymentConfig --> **not okay because invalid deployment-trigger from S2I ImageStream (must be removed)**

```yaml
...
apiVersion: "apps.openshift.io/v1"
kind: "DeploymentConfig"
...
spec:
  ...
  triggers:
  - imageChangeParams:
      automatic: true
      containerNames:
      - "quarkus-openshift-deployment"
      from:
        kind: "ImageStreamTag"
        name: "quarkus-openshift-deployment:1.0.0-SNAPSHOT"
    type: "ImageChange"
...
```

7) Route --> okay

**Expected behavior**

The combination of the extensions **quarkus-kubernetes** and **quarkus-container-image-docker** for **deployment-target=openshift** should build a container-image, push it to my docker-registry and then deploy only the needed OpenShift resources with a reference to the container-image in the docker-registry.

**Actual behavior**

The generated openshift.yml file contains some not needed resources and wrong settings for S2I build (S2I is not in place in my configuration). This prevents the automatic deployment to OpenShift without using S2I.

**To Reproduce**

1. Bootstrap a new quarkus maven project with the extensions: resteasy-jsonb, smallrye-health, container-image-docker, kubernetes
2. Add these properties to the application.properties 
```
quarkus.kubernetes-client.trust-certs=true
quarkus.kubernetes.deployment-target=openshift
quarkus.openshift.expose=true
quarkus.openshift.labels.app=quarkus-demo
```
3. Login to your OpenShift cluster `oc login https://my-openshift-cluster --token=...`
4. Login to your docker-registry `docker login ...`
5. Start build with deployment `mvn clean package -Dquarkus.kubernetes.deploy=true`
6. Open OpenShift WebConsole to see that deployment was not successful due to reported bug

**Environment**
- Quarkus version: 1.5.1-Final
- Build tool: Maven 3.6.3
- OpenShift 3.11

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #9663

### CODE REVIEW

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java [hunk 1]
```diff
@@ -69,6 +69,13 @@
     @ConfigItem
     public boolean push;
 
+    /**
+     * The name of the container image extension to use (e.g. docker, jib, s2i).
+     * The option will be used in case multiple extensions are present.
```
Comment (geoand) [inline extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java:74]: I think we should also update the container image documentation to mention this.

**Review by geoand — COMMENTED** [review]
Aren't there also some checks in the Kubernetes processor that need to be updated?
Or do those just check for the capabilities that you know only enable conditionally?

**Comment by iocanel** [comment]
> Aren't there also some checks in the Kubernetes processor that need to be updated?
> Or do those just check for the capabilities that you know only enable conditionally?

I don't think that anything else needs updating. But I might miss something.

**Comment by geoand** [comment]
I'll need to take another look tomorrow

**Review by geoand — APPROVED** [review]
Looks great, thanks!

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java (hunk 1 above)
Comment (geoand) [reply extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java:74]: Do you plan to add this in a next commit?

File: docs/src/main/asciidoc/container-image.adoc [hunk 2]
```diff
@@ -91,6 +91,17 @@ To push a container image for your project, `quarkus.container-image.push=true`
 
 NOTE: If no registry is set (using `quarkus.container-image.registry`) then `docker.io` will be used as the default.
 
+== Selecting among multiple extensions
+
+It does make sense to use multiple extension as part of the same build. When multiple container image extensions are present an error will be raised to inform the user. The user can either remove the uneeded extensions or select one using `application.properties`.
```
Comment (geoand) [inline docs/src/main/asciidoc/container-image.adoc:96]:
```suggestion
It does not make sense to use multiple extension as part of the same build. When multiple container image extensions are present an error will be raised to inform the user. The user can either remove the uneeded extensions or select one using `application.properties`.
```

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java (hunk 1 above)
Comment (iocanel) [reply extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java:74]: Just added the extra doc.
Also updated the error message that is displayed when there are conflicitng container image extension to inform the user about the `quarkus.container-image.builer` property.

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Seems like you need to reformat the code

**Comment by gsmet** [comment]
Pushed a rebase and fixed the formatting issue.

**Comment by geoand** [comment]
Thanks!

**Comment by geoand** [comment]
JVM tests are failing so I guess there is some problem with the PR

**Comment by geoand** [comment]
@iocanel do you plan to fix this?

**Pull request merged**